### ISSUE TITLE
Separate the reading of ww3_shel.inp 

### ISSUE THREAD

**Issue by JessicaMeixner-NOAA** [NOAA-EMC/WW3#679]
**Is your feature request related to a problem? Please describe.**
Create a separate routine to read ww3_shel.inp so that both ww3_shel program and the mesh based cap can use the same routine for reading the ww3_shel.inp file (and ww3_shel.nml).  

**Describe the solution you'd like**
New subroutine to read ww3_shel.inp to reduce repeated code, which will help with code maintenance. 

**Describe alternatives you've considered**
All initialization needs for ww3_shel.inp could be moved into a subroutine. 

**Additional context**
This is part of merging https://github.com/NOAA-EMC/WW3/pull/641 to the develop branch

**Comment by arunchawla-NOAA** [comment]
There is already a function to read ww_shel.inp data in wave_shel_inp.F90, is the request to use this function in ww3_shel.F90?

**Comment by JessicaMeixner-NOAA** [comment]
@arunchawla-NOAA yes, this way we don't have duplicative code and if code needs updated its done once for both places.

**Comment by MatthewMasarik-NOAA** [comment]
@arunchawla-NOAA, this has been my task.  I've started it but was pulled to higher priority items.  It will take a small dedicated  chunk of time that I can move through the duplicative regions and refactor the duplicated code out.
